core/src/main/java/org/mule/runtime/core/api/type/catalog/ApplicationTypeLoader.java

```diff
@@ -36,6 +38,8 @@
 public class ApplicationTypeLoader implements TypeLoader {
```

Comment:
per today's conversation:

* Rename to `ArtifactTypeLoader` (or better idea)
* Separate interface and impl

Reply:
Shouldn't we address that on a separate PR?